python/cudf/cudf/core/index.py

```diff
@@ -204,44 +203,36 @@ class RangeIndex(BaseIndex, BinaryOperand):
     def __init__(
         self, start, stop=None, step=1, dtype=None, copy=False, name=None
     ):
-        if step == 0:
-            raise ValueError("Step must not be zero.")
         if not cudf.api.types.is_hashable(name):
             raise ValueError("Name must be a hashable value.")
-        if dtype is not None and not is_signed_integer_dtype(dtype):
+        self._name = name
+        if dtype is not None and cudf.dtype(dtype).kind != "i":
             raise ValueError(f"{dtype=} must be a signed integer type")
 
         if isinstance(start, range):
-            therange = start
-            start = therange.start
-            stop = therange.stop
-            step = therange.step
-        if stop is None:
-            start, stop = 0, start
-        if not is_integer(start):
-            raise TypeError(
-                f"start must be an integer, not {type(start).__name__}"
-            )
-        self._start = int(start)
-        if not is_integer(stop):
-            raise TypeError(
-                f"stop must be an integer, not {type(stop).__name__}"
-            )
-        self._stop = int(stop)
-        if step is not None:
-            if not is_integer(step):
+            self._range = start
+        else:
+            if stop is None:
+                start, stop = 0, start
+            if not is_integer(start):
                 raise TypeError(
-                    f"step must be an integer, not {type(step).__name__}"
+                    f"start must be an integer, not {type(start).__name__}"
                 )
-            self._step = int(step)
-        else:
-            self._step = 1
-        self._index = None
-        self._name = name
-        self._range = range(self._start, self._stop, self._step)
-        # _end is the actual last element of RangeIndex,
-        # whereas _stop is an upper bound.
-        self._end = self._start + self._step * (len(self._range) - 1)
+            start = int(start)
+            if not is_integer(stop):
+                raise TypeError(
+                    f"stop must be an integer, not {type(stop).__name__}"
+                )
+            stop = int(stop)
+            if step is not None:
+                if not is_integer(step):
+                    raise TypeError(
+                        f"step must be an integer, not {type(step).__name__}"
+                    )
+                step = int(step)
```

Comment:
Can we extract this into a small helper function since you do it for start, stop, and step?

Reply:
Good idea. Made a validation helper function for this